File: php/features.php
<?php
declare(strict_types=1);

// IndoorEnvironmentalMonitoring SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IndoorEnvironmentalMonitoringFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IndoorEnvironmentalMonitoringBaseFeature();
            case "test":
                return new IndoorEnvironmentalMonitoringTestFeature();
            default:
                return new IndoorEnvironmentalMonitoringBaseFeature();
        }
    }
}

File: py/features.py
# IndoorEnvironmentalMonitoring SDK feature factory

from feature.base_feature import IndoorEnvironmentalMonitoringBaseFeature
from feature.test_feature import IndoorEnvironmentalMonitoringTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IndoorEnvironmentalMonitoringBaseFeature(),
        "test": lambda: IndoorEnvironmentalMonitoringTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
